<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Alexis Shulga, Brianna Correa, Marguerite Butler">

<title>Asterophryinae Microbiome Bibliography</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="microbiome_refs_files/libs/clipboard/clipboard.min.js"></script>
<script src="microbiome_refs_files/libs/quarto-html/quarto.js"></script>
<script src="microbiome_refs_files/libs/quarto-html/popper.min.js"></script>
<script src="microbiome_refs_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="microbiome_refs_files/libs/quarto-html/anchor.min.js"></script>
<link href="microbiome_refs_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="microbiome_refs_files/libs/quarto-html/quarto-syntax-highlighting-86daaaaad7353f9cc0c554efc1dd6d94.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="microbiome_refs_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="microbiome_refs_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="microbiome_refs_files/libs/bootstrap/bootstrap-561c0044c16d05b43a520218c0aaa88c.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
<div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="microbiome_refs.docx"><i class="bi bi-file-word"></i>MS Word</a></li><li><a href="microbiome_refs.pdf"><i class="bi bi-file-pdf"></i>PDF</a></li></ul></div></div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Asterophryinae Microbiome Bibliography</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Alexis Shulga, Brianna Correa, Marguerite Butler </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<section id="skin-microbiome" class="level2">
<h2 class="anchored" data-anchor-id="skin-microbiome">Skin Microbiome</h2>
<p><span class="citation" data-cites="Bierlich2018">Bierlich, K. C., Miller, C., DeForce, E., Friedlaender, A. S., Johnston, D. W., &amp; Apprill, A. (2018). Temporal and regional variability in the skin microbiome of humpback whales along the western antarctic peninsula. <em>Applied and Environmental Microbiology</em>, <em>84</em>(5), e02574–17. <a href="https://doi.org/10.1128/AEM.02574-17">https://doi.org/10.1128/AEM.02574-17</a></span></p>
<p>Humpback Whale study about seasonal changes with skin microbiome. Calf to Mother interaction. Microbial diversity increases during foraging season.</p>
<p><span class="citation" data-cites="Callewaert2020">Callewaert, C., Ravard Helffer, K., &amp; Lebaron, P. (2020). Skin microbiome and its interplay with the environment. <em>American Journal of Clinical Dermatology</em>, <em>21</em>, 4–11.</span></p>
<p>sequencing, bioinformatics, and multidisciplinary research are transforming our understanding of skin-associated microbial communities across diverse environments and species. It explores how microbes colonize the skin of humans, large mammals like humpback whales, and aquatic organisms, while emphasizing the influence of environmental factors such as water and air.</p>
<p><span class="citation" data-cites="Larsen2013">Larsen, A., Tao, Z., Bullard, S. A., &amp; Arias, C. R. (2013). Diversity of the skin microbiota of fishes: Evidence for host species specificity. <em>FEMS Microbiology Ecology</em>, <em>85</em>, 483–494.</span></p>
<p>Skin microbiota of fishes good oints towards host species specificity. extensive detail on cr and sequencing and data analysis of what can be done. interesting concet of the donminant microbiota. discusses seasonal changes.</p>
<p><span class="citation" data-cites="Kruger2020">Kruger, A. (2020). Frog skin microbiota vary with host species and environment but not chytrid infection. <em>Frontiers in Microbiology</em>, <em>11</em>, 1330. <a href="https://doi.org/10.3389/fmicb.2020.01330">https://doi.org/10.3389/fmicb.2020.01330</a></span></p>
<ul>
<li>frog skin microbiota vary by host species and environment, but not chytrid infection, highlighting the role of microbial composition in disease dynamics and resistance.</li>
</ul>
<p><span class="citation" data-cites="McCormack2018">McCormack, J. E., Rodriguez-Gomez, F., Tsai, W. L. E., Faircloth, B. C., &amp; Weckstein, J. D. (2018). Transforming museum specimens into genomic resources. In M. S. Webster (Ed.), <em>The extended specimen: Emerging frontiers in collections-based ornithological research</em> (p. 14). Taylor &amp; Francis. <a href="https://doi.org/10.1201/9781315120454">https://doi.org/10.1201/9781315120454</a></span></p>
<ul>
<li>Skin microbiome study using feather tip and preserved skin, slight destructive sampling on museum sample. Focused on western scrub jays. Used SNPs. Proved sampling can be long term success in museum samples.</li>
</ul>
<p><span class="citation" data-cites="Sylvain2020">Sylvain, F.-É., Holland, A., Bouslama, S., Audet-Gilbert, Émie, Lavoie, C., Val, A. L., &amp; Derome, N. (2020). Fish skin and gut microbiomes show contrasting signatures of host species and habitat. <em>Applied and Environmental Microbiology</em>, <em>86</em>(16), e00789–20. <a href="https://doi.org/10.1128/AEM.00789-20">https://doi.org/10.1128/AEM.00789-20</a></span></p>
<p>gut microbiome of teleost fish is shaped by species-specific factors, while the skin mucus microbiome is more influenced by environmental conditions, making it a potential biomarker for ecosystem changes.</p>
<p><span class="citation" data-cites="Tarnecki2013">Tarnecki, A., Tao, Z., &amp; Arias, C. (2013). Diversity of the skin microbiota of fishes: Evidence for host species specificity. <em>FEMS Microbiology Ecology</em>, <em>85</em>. <a href="https://doi.org/10.1111/1574-6941.12136">https://doi.org/10.1111/1574-6941.12136</a></span></p>
<ul>
<li>The study found that the skin microbiota of fish is highly species-specific, with Proteobacteria being the most abundant group. Certain bacteria, like Aeribacillus, were common across all species, while others, such as Neorickettsia and Microbacterium, were unique to specific fish species.</li>
</ul>
<p><span class="citation" data-cites="Zheng2022">Zheng, Y., Shi, J., Chen, Q., Deng, C., Yang, F., &amp; Wang, Y. (2022). Identifying individual-specific microbial DNA fingerprints from skin microbiomes. <em>FRONTIERS IN MICROBIOLOGY</em>, <em>13</em>. <a href="https://doi.org/10.3389/fmicb.2022.960043">https://doi.org/10.3389/fmicb.2022.960043</a></span></p>
<p>-Human study ID’s individual-specific microbial DNA fingerprints from skin microbiomes using metagenomic sequencing data. These fingerprints are stable over time, independent of body site, and can accurately identify donors with 89.78% accuracy. The analysis revealed that 10 out of 12 individual-specific fingerprints could be aligned to Cutibacterium acnes, proving the potential of these microbial markers for personal identification.</p>
<p><span class="citation" data-cites="Skowron2021">Skowron, K., Bauza-Kaszewska, J., Kraszewska, Z., Wiktorczyk-Kapischke, N., Grudlewska-Buda, K., Kwiecińska-Piróg, J., Wałecka-Zacharska, E., Radtke, L., &amp; Gospodarek-Komkowska, E. (2021). Human skin microbiome: Impact of intrinsic and extrinsic factors on skin microbiota. <em>Microorganisms</em>, <em>9</em>.</span></p>
<p>study of human skin microbiome in maintaining skin health, emphasizing how its complex microbial community—including bacteria, fungi, viruses, archaea, mites, and phages—interacts with the immune system and responds to internal and external environmental factors. It highlights the impact of microbiome imbalances (dysbiosis) on skin diseases, discusses molecular identification techniques for studying these microorganisms, and underscores the broader systemic consequences of skin microbiome disturbances beyond dermatological conditions</p>
</section>
<section id="gut-microbiome" class="level2">
<h2 class="anchored" data-anchor-id="gut-microbiome">Gut Microbiome</h2>
<p><span class="citation" data-cites="Chalifour2022">Chalifour, B. N., Elder, L. E., &amp; Li, J. (2022). Gut microbiome of century-old snail specimens stable across time in preservation. <em>Microbiome</em>, <em>10</em>(1), 99. <a href="https://doi.org/10.1186/s40168-022-01286-z">https://doi.org/10.1186/s40168-022-01286-z</a></span></p>
<ul>
<li>Century old specimen successful gutmicrobe extraction w/o destructive sampling. Methods:</li>
<li>Alcohol specimens. No problem obtaining gut microbiome from century old snail specimens, but they were never formalin fixed.</li>
<li>Location of collection more sig factor in gut microbiome composition of communities then what era snail came from (didn’t compare species to species). “The effects of decades-long museum preservation on host-microbial communities have not been systematically assessed.”</li>
</ul>
<p>##Formalin Fixed Sample DNA Collection</p>
<p><span class="citation" data-cites="Ramesh2017">Ramesh, G., Katiyar, A., Sujatha, R., Raj, A., Gupta, B., &amp; Kumar, A. (2017). Detection of microorganisms on formalin-fixed and stored pathology tissues: A microbiological study. <em>Journal of Oral and Maxillofacial Pathology : JOMFP</em>, <em>21</em>, 64–69. <a href="https://doi.org/10.4103/0973-029x.203788">https://doi.org/10.4103/0973-029x.203788</a></span></p>
<ul>
<li>Successful detect of formalin-fix tissue; destructive sampling; able to successfully sequence.</li>
</ul>
</section>
<section id="methodology-papers" class="level2">
<h2 class="anchored" data-anchor-id="methodology-papers">Methodology Papers</h2>
<p><span class="citation" data-cites="FieschiMeric2023"><strong>FieschiMeric2023?</strong></span></p>
<ul>
<li>Non-invasive swabbing technique for sampling salamanders to collect skin microbiome</li>
</ul>
</section>
<section id="environmental-factors" class="level2">
<h2 class="anchored" data-anchor-id="environmental-factors">Environmental Factors</h2>
</section>
<section id="community" class="level2">
<h2 class="anchored" data-anchor-id="community">Community</h2>
</section>
<section id="museum-and-historical-dna" class="level2">
<h2 class="anchored" data-anchor-id="museum-and-historical-dna">Museum and Historical DNA</h2>
<p><span class="citation" data-cites="Wandeler2007">Wandeler, P., Hoeck, P. E. A., &amp; Keller, L. F. (2007). Back to the future: Museum specimens in population genetics. <em>Trends in Ecology &amp; Evolution</em>, <em>22</em>(12), 634–642. https://doi.org/<a href="https://doi.org/10.1016/j.tree.2007.08.017">https://doi.org/10.1016/j.tree.2007.08.017</a></span></p>
<ul>
<li>Guidelines for working with degraded DNA from museum specimens, including tips on selecting samples, optimizing extraction methods, and using short PCR amplicons for better amplification. It also suggests sequencing strategies and bioinformatics tools designed to handle fragmented DNA, helping researchers unlock valuable genetic data from older specimens.</li>
</ul>
<p><span class="citation" data-cites="Heyn2010">Heyn, P., Stenzel, U., Briggs, A. W., Kircher, M., Hofreiter, M., &amp; Meyer, M. (2010). Road blocks on paleogenomes—polymerase extension profiling reveals the frequency of blocking lesions in ancient DNA. <em>Nucleic Acids Research</em>, <em>38</em>, e161–e161. <a href="https://api.semanticscholar.org/CorpusID:11149446">https://api.semanticscholar.org/CorpusID:11149446</a></span></p>
<ul>
<li>Issues with using aDNA. Good methods paper. Look into blocking lesions menionted because they were able to work around them but I want to understand more there.</li>
</ul>
<p><span class="citation" data-cites="Heindler2018">Heindler, F. M., Christiansen, H., Frédérich, B., Dettaı̈, A., Lepoint, G., Maes, G. E., Van de Putte, A. P., &amp; Volckaert, F. A. M. (2018). Historical DNA metabarcoding of the prey and microbiome of trematomid fishes using museum samples. <em>Frontiers in Ecology and Evolution</em>, <em>6</em>. <a href="https://doi.org/10.3389/fevo.2018.00151">https://doi.org/10.3389/fevo.2018.00151</a></span></p>
<ul>
<li><strong>Gut microbiome and prey identification from fish that have been formalin fixed.</strong> Between 20 and 100 years old. Sequenced cytochrome c oxidase subunit I (COI) 313bp from vertebrates and 16S rRNA from bacteria 450bp but had large dropout rates. Did not keep any COI, but able to characterize microbiomes from <strong>26 samples out of 225 starting samples</strong>.</li>
</ul>
<p><span class="citation" data-cites="Madison2023">Madison, B. C. A. W., Joseph D. AND LaBumbard. (2023). Shotgun metagenomics captures more microbial diversity than targeted 16S rRNA gene sequencing for field specimens and preserved museum specimens. <em>PLOS ONE</em>, <em>18</em>(9), 1–18. <a href="https://doi.org/10.1371/journal.pone.0291540">https://doi.org/10.1371/journal.pone.0291540</a></span></p>
<ul>
<li>Comparison of <strong>16S rRNA gene sequencing</strong> and <strong>short-read shotgun metagenomics</strong> methods for examining museum specimen-associated gut microbiomes.</li>
</ul>
<p><span class="citation" data-cites="Raxworthy2021">Raxworthy, C. J., &amp; Smith, B. T. (2021). Mining museums for historical DNA: Advances and challenges in museomics [Doi: 10.1016/j.tree.2021.07.009]. <em>Trends in Ecology &amp; Evolution</em>, <em>36</em>(11), 1049–1060. <a href="https://doi.org/10.1016/j.tree.2021.07.009">https://doi.org/10.1016/j.tree.2021.07.009</a></span></p>
<ul>
<li>Review of challenges with and recent advances in obtaining DNA sequence data from historical (museum) specimens.</li>
</ul>
<p><span class="citation" data-cites="Hykin2015">Hykin, K. A. M., Sarah M. AND Bi. (2015). Fixing formalin: A method to recover genomic-scale DNA sequence data from formalin-fixed museum specimens using high-throughput sequencing. <em>PLOS ONE</em>, <em>10</em>(10), 1–16. <a href="https://doi.org/10.1371/journal.pone.0141579">https://doi.org/10.1371/journal.pone.0141579</a></span></p>
<ul>
<li><p>Comparison of DNA sequencing of formalin-fixed <em>Anolis carolinensis</em> museum specimens 30 and 100 years old. Used Illumina high throughput sequencing to obtain massive amounts of short reads. Compared liver, leg muscle, and tail tips. Took care to avoid contamination, best yield was from phenol-chloriform protcol. Minimal PCR cycles. Used 100-bp paired-end Illumina sequencing, pooling both samples on one lane of a HiSeq2000. The older sample failed to provide usable data. For the younger sample, nuclear data did not have sufficient read depth, but they were able to get the whole mitochondrial genome (aided by the high copy number).</p></li>
<li><p>Three tissue types subsampled from formalin-fixed specimens of Anolis carolinensis, After trimming, successfully mapped 27.93% of the cleaned reads to the reference genome, were able to reconstruct the complete mitochondrial genome, and recovered an accurate phylogenetic placement for our specimen. Conclude amount of DNA available, which can vary depending on specimen age and preservation conditions</p></li>
</ul>
<p><span class="citation" data-cites="Yeates2016">Yeates, D. K., Zwick, A., &amp; Mikheyev, A. S. (2016). Museums are biobanks: Unlocking the genetic potential of the three billion specimens in the world’s biological collections. <em>Current Opinion in Insect Science</em>, <em>18</em>, 83–88. https://doi.org/<a href="https://doi.org/10.1016/j.cois.2016.09.009">https://doi.org/10.1016/j.cois.2016.09.009</a></span></p>
<ul>
<li>One of the earlier papers reviewing the issues with museum specimens as DNA sources. Focuses on the promise of high throughput methods. Short and sweet.</li>
</ul>
<p><span class="citation" data-cites="Zimmermann2008">Zimmermann, J., Hajibabaei, M., Blackburn, D. C., Hanken, J., Cantin, E., Posfai, J., &amp; Evans, T. C. (2008). DNA damage in preserved specimens and tissue samples: A molecular assessment. <em>Frontiers in Zoology</em>, <em>5</em>(1), 18. <a href="https://doi.org/10.1186/1742-9994-5-18">https://doi.org/10.1186/1742-9994-5-18</a></span></p>
<ul>
<li>A sobering accounting of the DNA damage induced by formalin fixation. They assess average fragment length as a function of age, it is around 50-70bp, with younger specimens longer.</li>
</ul>
<p><span class="citation" data-cites="Sampaio2023">Sampaio, F. L., Day, J. J., Mendis Wickramasinghe, L. J., Cyriac, V. P., Papadopoulou, A., Brace, S., Rajendran, A., Simon-Nutbrown, C., Flouris, T., Kapli, P., Ranga Vidanapathirana, D., Kotharambath, R., Kodandaramaiah, U., &amp; Gower, D. J. (2023). A near-complete species-level phylogeny of uropeltid snakes harnessing historical museum collections as a DNA source. <em>Molecular Phylogenetics and Evolution</em>, <em>178</em>, 107651. https://doi.org/<a href="https://doi.org/10.1016/j.ympev.2022.107651">https://doi.org/10.1016/j.ympev.2022.107651</a></span></p>
<ul>
<li>Phylogenetic study conducted using uropeltid snakes and hDNA. Able to work with swabs of nondestructive and destructive DNA sampling.</li>
</ul>
<p><span class="citation" data-cites="Paeaebo2004">Pääbo, S., Poinar, H., Serre, D., Jaenicke-Despres, V., Hebler, J., Rohland, N., Kuch, M., Krause, J., Vigilant, L., &amp; Hofreiter, M. (2004). Genetic analyses from ancient DNA. <em>Annual Review of Genetics</em>, <em>38</em>, 645–679. <a href="https://doi.org/10.1146/annurev.genet.37.110801.143214">https://doi.org/10.1146/annurev.genet.37.110801.143214</a></span></p>
<ul>
<li>ancient DNA vs h DNA and the advances made in detecting both. Application. Parent paper to Zimmermann 2008.</li>
</ul>
<p><span class="citation" data-cites="Staats2013">Staats, R. H. J. A. van de V., Martijn AND Erkens. (2013). Genomic treasure troves: Complete genome sequencing of herbarium and insect museum specimens. <em>PLOS ONE</em>, <em>8</em>(7), 1–11. <a href="https://doi.org/10.1371/journal.pone.0069189">https://doi.org/10.1371/journal.pone.0069189</a></span></p>
<ul>
<li>Using Beetle soaking method found in <span class="citation" data-cites="Heindler2018">Heindler, F. M., Christiansen, H., Frédérich, B., Dettaı̈, A., Lepoint, G., Maes, G. E., Van de Putte, A. P., &amp; Volckaert, F. A. M. (2018). Historical DNA metabarcoding of the prey and microbiome of trematomid fishes using museum samples. <em>Frontiers in Ecology and Evolution</em>, <em>6</em>. <a href="https://doi.org/10.3389/fevo.2018.00151">https://doi.org/10.3389/fevo.2018.00151</a></span>. Detailed how genomic sequencing was used on non-destructive sampling of items and what info was able to be gained off non-destructive/desicated plant samples</li>
</ul>
<p><span class="citation" data-cites="Raxworthy2021">Raxworthy, C. J., &amp; Smith, B. T. (2021). Mining museums for historical DNA: Advances and challenges in museomics [Doi: 10.1016/j.tree.2021.07.009]. <em>Trends in Ecology &amp; Evolution</em>, <em>36</em>(11), 1049–1060. <a href="https://doi.org/10.1016/j.tree.2021.07.009">https://doi.org/10.1016/j.tree.2021.07.009</a></span></p>
<p>-hDNA paper concerning advances in genomic studies in museum samples; useful for work cited paper and review summary of where we are somewhat currently</p>
<p><span class="citation" data-cites="Tin2014">Tin, M. M.-Y., Economo, E. P., &amp; Mikheyev, A. S. (2014). Sequencing degraded DNA from non-destructively sampled museum specimens for RAD-tagging and low-coverage shotgun phylogenetics. <em>PloS One</em>, <em>9</em>, e96793. <a href="https://doi.org/10.1371/journal.pone.0096793">https://doi.org/10.1371/journal.pone.0096793</a></span></p>
<ul>
<li>Addresses working with degraded DNA using RAD Tag system, helpful for methods should contamination be high</li>
</ul>
<p><span class="citation" data-cites="Madison2023">Madison, B. C. A. W., Joseph D. AND LaBumbard. (2023). Shotgun metagenomics captures more microbial diversity than targeted 16S rRNA gene sequencing for field specimens and preserved museum specimens. <em>PLOS ONE</em>, <em>18</em>(9), 1–18. <a href="https://doi.org/10.1371/journal.pone.0291540">https://doi.org/10.1371/journal.pone.0291540</a></span></p>
<ul>
<li>Shotgun metagenomic to capture microbial diversity in museum specimens theat were fixed in formalin, utilized destructive sampling</li>
</ul>
</section>
<section id="helpful-guide-to-format-annotated-bibliography-or-a-cv" class="level2">
<h2 class="anchored" data-anchor-id="helpful-guide-to-format-annotated-bibliography-or-a-cv">Helpful guide to format annotated bibliography (or a CV):</h2>
<p><span class="citation" data-cites="Heiss:2023">Heiss, A. (2023, January 9). <em>One <span>Simple</span> <span>Trick</span><sup>TM</sup> to Create Inline Bibliography Entries with <span>Markdown</span> and Pandoc</em>. <a href="https://doi.org/10.59350/hwwgk-v9636">https://doi.org/10.59350/hwwgk-v9636</a></span></p>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
});
</script>
</div> <!-- /content -->




</body></html>