---
title: "Asterophryinae Microbiome Bibliography"
author: "Alexis Shulga, Brianna Correa, Marguerite Butler"

format:
  html:
    toc: false
    number-sections: false
    highlight-style: github
  docx: 
    toc: false
    number-sections: false
    highlight-style: github
  pdf: 
    toc: false
    number-sections: false
    highlight-style: github
bibliography: microbiome.bib
csl: apa-cv.csl
suppress-bibliography: true
link-citations: false
citations-hover: false

---

## Skin Microbiome

[@Bierlich2018]

Humpback Whale study about seasonal changes with skin microbiome. Calf to Mother interaction. Microbial diversity increases during foraging season.

[@Callewaert2020]

sequencing, bioinformatics, and multidisciplinary research are transforming our understanding of skin-associated microbial communities across diverse environments and species. It explores how microbes colonize the skin of humans, large mammals like humpback whales, and aquatic organisms, while emphasizing the influence of environmental factors such as water and air.

[@Larsen2013]

Skin microbiota of fishes good oints towards host species specificity. extensive detail on cr and sequencing and data analysis of what can be done. interesting concet of the donminant microbiota. discusses seasonal changes.

[@Kruger2020]

-  frog skin microbiota vary by host species and environment, but not chytrid infection, highlighting the role of microbial composition in disease dynamics and resistance.

[@McCormack2018]

- Skin microbiome study using feather tip and preserved skin, slight destructive sampling on museum sample. Focused on western scrub jays. Used SNPs. Proved sampling can be long term success in museum samples. 

[@Sylvain2020]

gut microbiome of teleost fish is shaped by species-specific factors, while the skin mucus microbiome is more influenced by environmental conditions, making it a potential biomarker for ecosystem changes.

[@Tarnecki2013]

- The study found that the skin microbiota of fish is highly species-specific, with Proteobacteria being the most abundant group. Certain bacteria, like Aeribacillus, were common across all species, while others, such as Neorickettsia and Microbacterium, were unique to specific fish species.

[@Zheng2022]

-Human study ID's individual-specific microbial DNA fingerprints from skin microbiomes using metagenomic sequencing data. These fingerprints are stable over time, independent of body site, and can accurately identify donors with 89.78% accuracy. The analysis revealed that 10 out of 12 individual-specific fingerprints could be aligned to Cutibacterium acnes, proving the potential of these microbial markers for personal identification.

[@Skowron2021]

study of human skin microbiome in maintaining skin health, emphasizing how its complex microbial community—including bacteria, fungi, viruses, archaea, mites, and phages—interacts with the immune system and responds to internal and external environmental factors. It highlights the impact of microbiome imbalances (dysbiosis) on skin diseases, discusses molecular identification techniques for studying these microorganisms, and underscores the broader systemic consequences of skin microbiome disturbances beyond dermatological conditions


## Gut Microbiome

[@Chalifour2022]

- Century old specimen successful gutmicrobe extraction w/o destructive sampling. Methods:
- Alcohol specimens. No problem obtaining gut microbiome from century old snail specimens, but they were never formalin fixed.
- Location of collection more sig factor in gut microbiome composition of communities then what era snail came from (didn't compare species to species). "The effects of decades-long museum preservation on host-microbial communities have not been systematically assessed."

##Formalin Fixed Sample DNA Collection

[@Ramesh2017]

- Successful detect of formalin-fix tissue; destructive sampling; able to successfully sequence.


 
## Environmental Factors


## Community 


## Museum and Historical DNA

[@Wandeler2007]

-  Guidelines for working with degraded DNA from museum specimens, including tips on selecting samples, optimizing extraction methods, and using short PCR amplicons for better amplification. It also suggests sequencing strategies and bioinformatics tools designed to handle fragmented DNA, helping researchers unlock valuable genetic data from older specimens.


[@Heyn2010]

- Issues with using aDNA. Good methods paper. Look into blocking lesions menionted because they were able to work around them but I want to understand more there. 

[@Heindler2018] 

-  __Gut microbiome and prey identification from fish that have been formalin fixed.__ Between 20 and 100 years old. Sequenced cytochrome c oxidase subunit I (COI) 313bp from vertebrates and 16S rRNA from bacteria 450bp but had large dropout rates. Did not keep any COI, but able to characterize microbiomes from __26 samples out of 225 starting samples__.

[@Madison2023]

-  Comparison of __16S rRNA gene sequencing__ and __short-read shotgun metagenomics__ methods for examining museum specimen-associated gut microbiomes.  

[@Raxworthy2021]

-  Review of challenges with and recent advances in obtaining DNA sequence data from historical (museum) specimens.

[@Hykin2015]

- Comparison of DNA sequencing of formalin-fixed _Anolis carolinensis_ museum specimens 30 and 100 years old. Used Illumina high throughput sequencing to obtain massive amounts of short reads. Compared liver, leg muscle, and tail tips.  Took care to avoid contamination, best yield was from phenol-chloriform protcol. Minimal PCR cycles. Used 100-bp paired-end Illumina sequencing, pooling both samples on one lane of a HiSeq2000. The older sample failed to provide usable data. For the younger sample, nuclear data did not have sufficient read depth, but they were able to get the whole mitochondrial genome (aided by the high copy number). 

- Three tissue types subsampled from formalin-fixed specimens of Anolis carolinensis, After trimming, successfully mapped 27.93% of the cleaned reads to the reference genome, were able to reconstruct the complete mitochondrial genome, and recovered an accurate phylogenetic placement for our specimen. Conclude amount of DNA available, which can vary depending on specimen age and preservation conditions

[@Yeates2016]

- One of the earlier papers reviewing the issues with museum specimens as DNA sources. Focuses on the promise of high throughput methods. Short and sweet. 

[@Zimmermann2008] 

- A sobering accounting of the DNA damage induced by formalin fixation. They assess average fragment length as a function of age, it is around 50-70bp, with younger specimens longer.

[@Sampaio2023]

- Phylogenetic study conducted using uropeltid snakes and hDNA. Able to work with swabs of nondestructive and destructive DNA sampling. 

[@Paeaebo2004]

- ancient DNA vs h DNA and the advances made in detecting both. Application. Parent paper to Zimmermann 2008.

[@Staats2013]

- Using Beetle soaking method found in [@Heindler2018]. Detailed how genomic sequencing was used on non-destructive sampling of items and what info was able to be gained off non-destructive/desicated plant samples

[@Raxworthy2021]

-hDNA paper concerning advances in genomic studies in museum samples; useful for work cited paper and review summary of where we are somewhat currently

[@Tin2014]

- Addresses working with degraded DNA using RAD Tag system, helpful for methods should contamination be high

[@Madison2023]

- Shotgun metagenomic to capture microbial diversity in museum specimens theat were fixed in formalin, utilized destructive sampling


## Helpful guide to format annotated bibliography (or a CV):   

[@Heiss:2023] 