<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Alexis Shulga, Brianna Correa, Marguerite Butler">

<title>Asterophryinae Skin Microbiome Project</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="urop-skin-microbiome-project_files/libs/clipboard/clipboard.min.js"></script>
<script src="urop-skin-microbiome-project_files/libs/quarto-html/quarto.js"></script>
<script src="urop-skin-microbiome-project_files/libs/quarto-html/popper.min.js"></script>
<script src="urop-skin-microbiome-project_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="urop-skin-microbiome-project_files/libs/quarto-html/anchor.min.js"></script>
<link href="urop-skin-microbiome-project_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="urop-skin-microbiome-project_files/libs/quarto-html/quarto-syntax-highlighting-86daaaaad7353f9cc0c554efc1dd6d94.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="urop-skin-microbiome-project_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="urop-skin-microbiome-project_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="urop-skin-microbiome-project_files/libs/bootstrap/bootstrap-6deea4c8f31ef074d8dc9809c87aa3b7.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
<div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="urop-skin-microbiome-project.docx"><i class="bi bi-file-word"></i>MS Word</a></li><li><a href="urop-skin-microbiome-project.pdf"><i class="bi bi-file-pdf"></i>PDF</a></li></ul></div></div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Asterophryinae Skin Microbiome Project</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Alexis Shulga, Brianna Correa, Marguerite Butler </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<section id="project-description-1500-character-limit" class="level2">
<h2 class="anchored" data-anchor-id="project-description-1500-character-limit">Project Description (1500 character limit)</h2>
<p>The skin and its microbiome are the first layer between an organism and its environment. In moist-skinned animals such as frogs, the skin mucosa is particularly sensitive to external factors which may vary by habitat or site. The microbiome may play an important functional role in mediating the interaction between the host and its environment <span class="citation" data-cites="Troitsky:2023">(<a href="#ref-Troitsky:2023" role="doc-biblioref">Troitsky et al., 2023</a>)</span>. Another intruiging role that the microbial community may play is providing some level of protection against pathogens, particularly fungal pathogens, including chytridiomycosis which is linked to mass extinction of many amphibian species <span class="citation" data-cites="Bates:2022">(<a href="#ref-Bates:2022" role="doc-biblioref">Bates et al., 2022</a>)</span>. Although skin microbiomes are being characterized, we still know very little about the causes of microbiome diversity, whether bacterial composition play some functional role, comes directly from the local environment, or is related to host diversity <span class="citation" data-cites="Sylvain:2020 Kruger:2020 Gajewski:2023 Garcia-Recinos:2019">(<a href="#ref-Gajewski:2023" role="doc-biblioref">Gajewski et al., 2023</a>; <a href="#ref-Garcia-Recinos:2019" role="doc-biblioref">Garcia-Recinos et al., 2019</a>; <a href="#ref-Kruger:2020" role="doc-biblioref">Kruger, 2020</a>; <a href="#ref-Sylvain:2020" role="doc-biblioref">Sylvain et al., 2020</a>)</span>.</p>
<p>In 2014, the Butler Lab conducted a field expedition to Papua New Guinea, one the worldʻs biggest hotspots for frog diversity and sampled closely related frogs at three sites that vary in ecology (terrestrial, shrub, stream), and live together in complex tropical communities <span class="citation" data-cites="Hill:2022">(<a href="#ref-Hill:2022" role="doc-biblioref">Hill et al., 2022</a>)</span>. Fortuitously, these frogs were carefully collected and individually stored in ethanol providing a unique dataset to explore factors which structure skin microbiome using NextGen sequencing metabarcoding methods. By comparing microbial community structure across sites, microhabitats, and species, I will test the hypotheses that microbiome composition is structured by microhabitat use <span class="citation" data-cites="Bierlich:2018">(<a href="#ref-Bierlich:2018" role="doc-biblioref">Bierlich et al., 2018</a>)</span>, influenced by functional benefit such as protection against pathogens <span class="citation" data-cites="Bletz:2017">(<a href="#ref-Bletz:2017" role="doc-biblioref">Bletz et al., 2017</a>)</span>, is site specific perhaps resulting from differences in the source population of microbes (e.g., soil microbial diversity), or is host specific <span class="citation" data-cites="Zheng:2022">(<a href="#ref-Zheng:2022" role="doc-biblioref">Zheng et al., 2022</a>)</span>.</p>
</section>
<section id="signficance-1500-character-limit" class="level2">
<h2 class="anchored" data-anchor-id="signficance-1500-character-limit">Signficance (1500 character limit)</h2>
<p>There are over 7000 species of frogs which are widely regarded as critical indicators of ecosystem health, as all amphibians have moist skin, are sensitive to environmental changes, and better to reveal complex skin microbiome dynamics.</p>
<p>Microbiome studies are rapidly growing in number yet we still know little about whether community composition is an accident of circumstance or whether it serves important functions. Are the skin bacteria and fungi simply a sample of the soil where they happen to live, or do the particular combination of microbes confer protection, such as inhibiting or outcompeting pathogens, anti-fungal protection, immune boosts <span class="citation" data-cites="Bates:2022">(<a href="#ref-Bates:2022" role="doc-biblioref">Bates et al., 2022</a>)</span>? Given the devastating role of emerging fungal pathogens <span class="citation" data-cites="Schilliger:2023">(<a href="#ref-Schilliger:2023" role="doc-biblioref">Schilliger et al., 2023</a>)</span>, understanding the relationships between microbial and fungal diversity are critical.</p>
<p>Alternatively do microbiomes mediate the absorption of water, nutrients, and functional exchanges with the environment <span class="citation" data-cites="Hughey:2017">(<a href="#ref-Hughey:2017" role="doc-biblioref">Hughey et al., 2017</a>)</span>? Comparing microbiomes of frogs that live together at the same site, but are found in the leaf litter vs.&nbsp;on plants vs.&nbsp;in the stream can reveal these associations <span class="citation" data-cites="Gajewski:2023 Garcia-Recinos:2019">(<a href="#ref-Gajewski:2023" role="doc-biblioref">Gajewski et al., 2023</a>; <a href="#ref-Garcia-Recinos:2019" role="doc-biblioref">Garcia-Recinos et al., 2019</a>)</span>, lessons which are applicable to all species.</p>
<p>Finally my study will provide an important baseline for understanding the relationship between skin and gut microbiome, and for future studies using museum specimens which have been formalin-fixed. While challenging to sequence, unlocking this “biobank” will provide access to thousands of important historical and geographical samples.</p>
</section>
<section id="processmethodology-1500-character-limit" class="level2">
<h2 class="anchored" data-anchor-id="processmethodology-1500-character-limit">Process/Methodology (1500 character limit)</h2>
<p>DNA will be isolated from sterile skin swabs of frog samples stored in ethanol at -20C in the Butler lab using Qiagen DNeasy kits. Bacterial 16S rRNA V4 region <span class="citation" data-cites="Madison:2023">(<a href="#ref-Madison:2023" role="doc-biblioref">Madison et al., 2023</a>)</span> and fungal ITS region<span class="citation" data-cites="Insuk:2024">(<a href="#ref-Insuk:2024" role="doc-biblioref">Insuk et al., 2024</a>)</span> sequences will be amplified using universal primers following the indicated protocols. Amplicons will be purified, unique barcodes attached to identify each frog sample, pooled, and sequenced following Next-Gen protocols using MinION Flongle <span class="citation" data-cites="Srivathsan:2021 Vasilita:2024">(<a href="#ref-Srivathsan:2021" role="doc-biblioref">Srivathsan et al., 2021</a>; <a href="#ref-Vasilita:2024" role="doc-biblioref">Vasilita et al., 2024</a>)</span>. ONTbarcoder will be used to generate demultiplexed, high-quality barcodes <span class="citation" data-cites="Srivathsan:2021">(<a href="#ref-Srivathsan:2021" role="doc-biblioref">Srivathsan et al., 2021</a>)</span>. Taxonomic assignments will be conducted via BLAST, and phylogenies constructed using IQTREE2.</p>
<p>Microbiome diversity and compositional variation across host species, microhabitats, and localities, will be analyzed using the vegan package in the R statistical computing environment. The hypotheses can be assessed in several ways: if microbiome diversity is structured by microhabitat, we should see similarity in microbial generic composition by host microhabitat (terrestrial vs.&nbsp;shrub vs.&nbsp;stream) rather than by host genus or by site <span class="citation" data-cites="Sylvain:2020">(e.g., <a href="#ref-Sylvain:2020" role="doc-biblioref">Sylvain et al., 2020</a>)</span>. Similarly, network cluster analysis should reveal which factor is dominant (microhabitat, site, or host genus). We will also assess the strength of correlations between microbial and fungal taxa, and dominance of taxa of interest.</p>
</section>
<section id="timetable-1500-character-limit" class="level2">
<h2 class="anchored" data-anchor-id="timetable-1500-character-limit">Timetable (1500 character limit)</h2>
<p>The project will begin Summer 2025 and will end Fall 2025. The applicant is currently training with the Butler Lab. The samples are in the lab and Alexis will have the necessary training in DNA extraction, PCR, and sequencing to start the experiments in this proposal in the Summer of 2025. Alexis will be here in Hawaii and have aqequate time to perform the experiments. Sample preparation and DNA extraction will be completed by June. Next-Gen sequencing will be completed in July, with bioinformatic analysis in August. We will simultaneously begin data science and analysis training in August to conduct microbime diversity and compositional analysis <span class="citation" data-cites="Butler:2023">Butler (<a href="#ref-Butler:2023a" role="doc-biblioref">2023b</a>)</span>. August and September will be spent producing analyses and figures to address hypotheses. The remaining time in October to Mid-November will allow for writing up the results in the form of the Project Completion report. Alexis will present a poster at the UH Manoa Undergraduate Research Showcase in December 2025.</p>
</section>
<section id="applicants-role-explain-your-responsibility-for-the-proposed-project-and-describe-the-expected-learning-outcomes.-explain-your-role-in-1-writing-the-proposed-project-2-carrying-out-the-proposed-project-and-3-describe-the-personal-professional-andor-academic-benefits-that-you-expect-to-gain-from-your-role-in-this-project.-1500-character-limit" class="level2">
<h2 class="anchored" data-anchor-id="applicants-role-explain-your-responsibility-for-the-proposed-project-and-describe-the-expected-learning-outcomes.-explain-your-role-in-1-writing-the-proposed-project-2-carrying-out-the-proposed-project-and-3-describe-the-personal-professional-andor-academic-benefits-that-you-expect-to-gain-from-your-role-in-this-project.-1500-character-limit">Applicant’s Role: Explain your responsibility for the proposed project and describe the expected learning outcomes. Explain your role in 1) writing the proposed project 2) carrying out the proposed project and 3) describe the personal, professional, and/or academic benefits that you expect to gain from your role in this project. (1500 Character Limit)</h2>
</section>
<section id="biographical-sketch-1500-character-limit" class="level2">
<h2 class="anchored" data-anchor-id="biographical-sketch-1500-character-limit">Biographical Sketch (1500 Character Limit)</h2>
<p>As a current freshman at the University of Hawaii at Manoa majoring in molecular cell biology, I carry a long-term goal of becoming a professor of environmental microbiology. As an early graduate valedictorian of Roosevelt high school and a current Dean’s List student with a 3.93 GPA, I have consistently demonstrated academic excellence. My passion for microbiomes and microorganisms was cultivated during a research internship at the Pacific Center for Molecular Biodiversity (PCMB) at the Bishop Museum my freshman year of high school. Here, I gained hands-on experience with microscopy, micropipettes, DNA extraction, PCR and DNA sequencing. Furthermore, I also gained knowledge with standard laboratory procedures such as data analysis such as calculating the amount of microorganisms and the species found throughout the Bishop Museum, this analysis was presented to researchers in the field and other Bishop Museum members with my PCMB intern group.</p>
<p>This research project resonates deeply with me because it integrates historical and modern microbiome analyses to explore how environmental factors have shaped the skin and gut microbiomes of microhylid frogs over 50 years. This project combines my love for museums, specifically the Bishop Museum, and research for micro biodiversity. As a future professor, investigating how different microhabitats (soil/leaf litter, shrubs, and streams) influence these communities will be significant topics of discussion for my future academic publications. Engaging in this project will not only enhance my technical and analytical skills, but also contribute significantly to my future research career and aspirations in academia.</p>
</section>
<section id="training-research-compliance-1500-character-limit" class="level2">
<h2 class="anchored" data-anchor-id="training-research-compliance-1500-character-limit">Training &amp; Research Compliance (1500 character limit):</h2>
<p>Alexis has joined the lab and participated in lab meetings. She completed the EHSO safety checklist on March 3, 2025 and copies of the paperwork are kept on file with the Butler Lab. During the training, hazards that may be present in the lab were described, among other topics covered by the checklist.</p>
<p>Alexis has completed EHSO general laboratory safety training on March 1, 2025, as required by the IBC. All genetic and biological safety aspects of the project are covered by Dr.&nbsp;Butlerʻs IBC protocol, which will be renewed with Alexis added before the announcement of UROP awards.</p>
<p>Project-specific training will be provided by Dr.&nbsp;Marguerite Butler beginning in March 2025. Dr.&nbsp;Butler is an expert in the methods described here and has protocol specific SOPs available for review prior to hands-on training. The lab has detailed emergency response and spill protocols that are reviewed before working in the lab. These discussions and training sessions have been ongoing since February 2025 to ensure all necessary compliance requirements are fully met by the start of the project.</p>
</section>
<section id="bibliographyreferences-5000-character-limit" class="level2 unnumbered">


</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">Bibliography/References (5000 Character Limit)</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" data-line-spacing="2" role="list">
<div id="ref-Bates:2022" class="csl-entry" role="listitem">
Bates, K. A., Sommer, U., Hopkins, K. P., Shelton, J. M. G., Wierzbicki, C., Sergeant, C., Tapley, B., Michaels, C. J., Schmeller, D. S., Loyau, A., Bosch, J., Viant, M. R., Harrison, X. A., Garner, T. W. J., &amp; Fisher, M. C. (2022). Microbiome function predicts amphibian chytridiomycosis disease dynamics [Article]. <em>MICROBIOME</em>, <em>10</em>(1). <a href="https://doi.org/10.1186/s40168-021-01215-6">https://doi.org/10.1186/s40168-021-01215-6</a>
</div>
<div id="ref-Bierlich:2018" class="csl-entry" role="listitem">
Bierlich, K. C., Miller, C., DeForce, E., Friedlaender, A. S., Johnston, D. W., &amp; Apprill, A. (2018). Temporal and regional variability in the skin microbiome of humpback whales along the western antarctic peninsula [Article]. <em>APPLIED AND ENVIRONMENTAL MICROBIOLOGY</em>, <em>84</em>(5). <a href="https://doi.org/10.1128/AEM.02574-17">https://doi.org/10.1128/AEM.02574-17</a>
</div>
<div id="ref-Bletz:2017" class="csl-entry" role="listitem">
Bletz, M. C., Perl, R. G. B., &amp; Vences, M. (2017). Skin microbiota differs drastically between co-occurring frogs and newts [Article]. <em>ROYAL SOCIETY OPEN SCIENCE</em>, <em>4</em>(4). <a href="https://doi.org/10.1098/rsos.170107">https://doi.org/10.1098/rsos.170107</a>
</div>
<div id="ref-Butler:2023" class="csl-entry" role="listitem">
Butler, M. A. (2023a). <em>Data science in r for biologists - course website</em>. <a href="https://mbutler808.github.io/rclass/">https://mbutler808.github.io/rclass/</a>
</div>
<div id="ref-Butler:2023a" class="csl-entry" role="listitem">
Butler, M. A. (2023b). <em>Phylogenetic inference with IQTREE2</em>. <a href="https://mbutler808.github.io/rclass/posts/2023-04-13-iqtree-phylogenetics/">https://mbutler808.github.io/rclass/posts/2023-04-13-iqtree-phylogenetics/</a>
</div>
<div id="ref-Gajewski:2023" class="csl-entry" role="listitem">
Gajewski, Z., Johnson, L. R., Medina, D., Crainer, W. W., Nagy, C. M., &amp; Belden, L. K. (2023). Skin bacterial community differences among three species of co-occurring ranid frogs [Article]. <em>PEERJ</em>, <em>11</em>. <a href="https://doi.org/10.7717/peerj.15556">https://doi.org/10.7717/peerj.15556</a>
</div>
<div id="ref-Garcia-Recinos:2019" class="csl-entry" role="listitem">
Garcia-Recinos, L., Burrowes, P. A., &amp; Dominguez-Bello, M. (2019). The skin microbiota of <em>eleutherodactylus</em> frogs: Effects of host ecology, phylogeny, and local environment [Article]. <em>FRONTIERS IN MICROBIOLOGY</em>, <em>10</em>. <a href="https://doi.org/10.3389/fmicb.2019.02571">https://doi.org/10.3389/fmicb.2019.02571</a>
</div>
<div id="ref-Hill:2022" class="csl-entry" role="listitem">
Hill, E. C., Fraser, C. J., Gao, D. F., Jarman, M. J., Henry, E. R., Iova, B., Allison, A., &amp; Butler, M. A. (2022). <span class="nocase">Resolving the deep phylogeny: Implications for early adaptive radiation, cryptic, and present-day ecological diversity of Papuan microhylid frogs</span>. <em><span>Molecular Phylogenetics and Evolution</span></em>, <em>177</em>, 107618. https://doi.org/<a href="https://doi.org/10.1016/j.ympev.2022.107618">https://doi.org/10.1016/j.ympev.2022.107618</a>
</div>
<div id="ref-Hughey:2017" class="csl-entry" role="listitem">
Hughey, M. C., Pena, J. A., Reyes, R., Medina, D., Belden, L. K., &amp; Burrowes, P. A. (2017). Skin bacterial microbiome of a generalist puerto rican frog varies along elevation and land use gradients [Article]. <em>PEERJ</em>, <em>5</em>. <a href="https://doi.org/10.7717/peerj.3688">https://doi.org/10.7717/peerj.3688</a>
</div>
<div id="ref-Insuk:2024" class="csl-entry" role="listitem">
Insuk, C., Cheeptham, N., Lausen, C., &amp; Xu, J. (2024). DNA metabarcoding analyses reveal fine-scale microbiome structures on western canadian bat wings [Article]. <em>MICROBIOLOGY SPECTRUM</em>, <em>12</em>(12). <a href="https://doi.org/10.1128/spectrum.00376-24">https://doi.org/10.1128/spectrum.00376-24</a>
</div>
<div id="ref-Kruger:2020" class="csl-entry" role="listitem">
Kruger, A. (2020). Frog skin microbiota vary with host species and environment but not chytrid infection [Article]. <em>FRONTIERS IN MICROBIOLOGY</em>, <em>11</em>. <a href="https://doi.org/10.3389/fmicb.2020.01330">https://doi.org/10.3389/fmicb.2020.01330</a>
</div>
<div id="ref-Madison:2023" class="csl-entry" role="listitem">
Madison, J. D., Labumbard, B. C., &amp; Woodhams, D. C. (2023). Shotgun metagenomics captures more microbial diversity than targeted 16S rRNA gene sequencing for field specimens and preserved museum specimens [Article]. <em>PLOS ONE</em>, <em>18</em>(9). <a href="https://doi.org/10.1371/journal.pone.0291540">https://doi.org/10.1371/journal.pone.0291540</a>
</div>
<div id="ref-Schilliger:2023" class="csl-entry" role="listitem">
Schilliger, L., Paillusseau, C., Francois, C., &amp; Bonwitt, J. (2023). Major emerging fungal diseases of reptiles and amphibians [Review]. <em>PATHOGENS</em>, <em>12</em>(3). <a href="https://doi.org/10.3390/pathogens12030429">https://doi.org/10.3390/pathogens12030429</a>
</div>
<div id="ref-Srivathsan:2021" class="csl-entry" role="listitem">
Srivathsan, A., Lee, L., Katoh, K., Hartop, E., Kutty, S. N., Wong, J., Yeo, D., &amp; Meier, R. (2021). ONTbarcoder and MinION barcodes aid biodiversity discovery and identification by everyone, for everyone [Article]. <em>BMC BIOLOGY</em>, <em>19</em>(1). <a href="https://doi.org/10.1186/s12915-021-01141-x">https://doi.org/10.1186/s12915-021-01141-x</a>
</div>
<div id="ref-Sylvain:2020" class="csl-entry" role="listitem">
Sylvain, F.-E., Holland, A., Bouslama, S., Audet-Gilbert, E., Lavoie, C., Val, A. L., &amp; Derome, N. (2020). Fish skin and gut microbiomes show contrasting signatures of host species and habitat [Article]. <em>APPLIED AND ENVIRONMENTAL MICROBIOLOGY</em>, <em>86</em>(16). <a href="https://doi.org/10.1128/AEM.00789-20">https://doi.org/10.1128/AEM.00789-20</a>
</div>
<div id="ref-Troitsky:2023" class="csl-entry" role="listitem">
Troitsky, T. S., Laine, V. N., &amp; Lilley, T. M. (2023). When the host’s away, the pathogen will play: The protective role of the skin microbiome during hibernation [Review]. <em>ANIMAL MICROBIOME</em>, <em>5</em>(1). <a href="https://doi.org/10.1186/s42523-023-00285-1">https://doi.org/10.1186/s42523-023-00285-1</a>
</div>
<div id="ref-Vasilita:2024" class="csl-entry" role="listitem">
Vasilita, C., Feng, V., Hansen, A. K., Hartop, E., Srivathsan, A., Struijk, R., &amp; Meier, R. (2024). Express barcoding with NextGenPCR and MinION for species-level sorting of ecological samples [Article]. <em>MOLECULAR ECOLOGY RESOURCES</em>, <em>24</em>(3). <a href="https://doi.org/10.1111/1755-0998.13922">https://doi.org/10.1111/1755-0998.13922</a>
</div>
<div id="ref-Zheng:2022" class="csl-entry" role="listitem">
Zheng, Y., Shi, J., Chen, Q., Deng, C., Yang, F., &amp; Wang, Y. (2022). Identifying individual-specific microbial DNA fingerprints from skin microbiomes [Article]. <em>FRONTIERS IN MICROBIOLOGY</em>, <em>13</em>. <a href="https://doi.org/10.3389/fmicb.2022.960043">https://doi.org/10.3389/fmicb.2022.960043</a>
</div>
</div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>