---
title: "Asterophryinae Skin Microbiome Project"
author: "Alexis Shulga, Brianna Correa, Marguerite Butler"

format:
  html:
    toc: false
    number-sections: false
    highlight-style: github
  docx: 
    toc: false
    number-sections: false
    highlight-style: github
  pdf: 
    toc: false
    number-sections: false
    highlight-style: github
    cite-method: natbib
    natbiboptions: numbers
bibliography: microbiome.bib
csl: apa.csl

---

## Project Description (1500 character limit)

The skin and its microbiome are the first layer between an organism and its environment. In moist-skinned animals such as frogs, the skin mucosa is particularly sensitive to external factors which may vary by habitat or site. The microbiome may play an important functional role in mediating the interaction between the host and its environment [@Troitsky:2023]. Another intruiging role that the microbial community may play is providing some level of protection against pathogens, particularly fungal pathogens, including chytridiomycosis which is linked to mass extinction of many amphibian species [@Bates:2022]. Although skin microbiomes are being characterized, we still know very little about the causes of microbiome diversity, whether bacterial composition play some functional role, comes directly from the local environment, or is related to host diversity [@Sylvain:2020;@Kruger:2020;@Gajewski:2023;@Garcia-Recinos:2019].  

In 2014, the Butler Lab conducted a field expedition to Papua New Guinea, one the worldʻs biggest hotspots for frog diversity and sampled closely related frogs at three sites that vary in ecology (terrestrial, shrub, stream), and live together in complex tropical communities [@Hill:2022].  Fortuitously, these frogs were carefully collected and individually stored in ethanol providing a unique dataset to explore factors which structure skin microbiome using NextGen sequencing metabarcoding methods. By comparing microbial community structure across sites, microhabitats, and species, I will test the hypotheses that microbiome composition is structured by microhabitat use [@Bierlich:2018], influenced by functional benefit such as protection against pathogens [@Bletz:2017], is site specific perhaps resulting from differences in the source population of microbes (e.g., soil microbial diversity), or is host specific [@Zheng:2022]. 

## Signficance (1500 character limit)


There are over 7000 species of frogs which are widely regarded as critical indicators of ecosystem health, as all amphibians have moist skin, are sensitive to environmental changes, and better to reveal complex skin microbiome dynamics. 

Microbiome studies are rapidly growing in number yet we still know little about whether community composition is an accident of circumstance or whether it serves important functions. Are the skin bacteria and fungi simply a sample of the soil where they happen to live, or do the particular combination of microbes confer protection, such as inhibiting or outcompeting pathogens, anti-fungal protection, immune boosts [@Bates:2022]?  Given the devastating role of emerging fungal pathogens [@Schilliger:2023], understanding the relationships between microbial and fungal diversity are critical. 

Alternatively do microbiomes mediate the absorption of water, nutrients, and functional exchanges with the environment [@Hughey:2017]? Comparing microbiomes of frogs that live together at the same site, but are found in the leaf litter vs. on plants vs. in the stream can reveal these associations [@Gajewski:2023;@Garcia-Recinos:2019], lessons which are applicable to all species. 

Finally my study will provide an important baseline for understanding the relationship between skin and gut microbiome, and for future studies using museum specimens which have been formalin-fixed. While challenging to sequence, unlocking this "biobank" will provide access to thousands of important historical and geographical samples. 


## Process/Methodology (1500 character limit)

DNA will be isolated from sterile skin swabs of frog samples stored in ethanol at -20C in the Butler lab using Qiagen DNeasy kits. Bacterial 16S rRNA V4 region [@Madison:2023] and fungal ITS region[@Insuk:2024] sequences will be amplified using universal primers following the indicated protocols. Amplicons will be purified, unique barcodes attached to identify each frog sample, pooled, and sequenced following Next-Gen protocols using MinION Flongle [@Srivathsan:2021;@Vasilita:2024]. ONTbarcoder will be used to generate demultiplexed, high-quality barcodes [@Srivathsan:2021].  Taxonomic assignments will be conducted via BLAST, and phylogenies constructed using IQTREE2. 

Microbiome diversity and compositional variation across host species, microhabitats, and localities, will be analyzed using the vegan package in the R statistical computing environment. The hypotheses can be assessed in several ways: if microbiome diversity is structured by microhabitat, we should see similarity in microbial generic composition by host microhabitat (terrestrial vs. shrub vs. stream) rather than by host genus or by site [e.g., @Sylvain:2020]. Similarly, network cluster analysis should reveal which factor is dominant (microhabitat, site, or host genus). We will also assess the strength of correlations between microbial and fungal taxa, and dominance of taxa of interest. 


## Timetable (1500 character limit)

The project will begin Summer 2025 and will end Fall 2025. The applicant is currently training with the Butler Lab. The samples are in the lab and Alexis will have the necessary training in DNA extraction, PCR, and sequencing to start the experiments in this proposal in the Summer of 2025. Alexis will be here in Hawaii and have aqequate time to perform the experiments. Sample preparation and DNA extraction will be completed by June. Next-Gen sequencing will be completed in July, with bioinformatic analysis in August. We will simultaneously begin data science and analysis training in August to conduct microbime diversity and compositional analysis [tutorials available from PI Butlerʻs course website e.g., @Butler:2023,@Butler:2023a]. August and September will be spent producing analyses and figures to address hypotheses. The remaining time in October to Mid-November will allow for writing up the results in the form of the Project Completion report. Alexis will present a poster at the UH Manoa Undergraduate Research Showcase in December 2025. 


## Applicant's Role: Explain your responsibility for the proposed project and describe the expected learning outcomes. Explain your role in 1) writing the proposed project 2) carrying out the proposed project and 3) describe the personal, professional, and/or academic benefits that you expect to gain from your role in this project. (1500 Character Limit) 


## Biographical Sketch (1500 Character Limit)

As a current freshman at the University of Hawaii at Manoa majoring in molecular cell biology, I carry a long-term goal of becoming a professor of environmental microbiology. As an early graduate valedictorian of Roosevelt high school and a current Dean’s List student with a 3.93 GPA, I have consistently demonstrated academic excellence. My passion for microbiomes and microorganisms was cultivated during a research internship at the Pacific Center for Molecular Biodiversity (PCMB) at the Bishop Museum my freshman year of high school. Here, I gained hands-on experience with microscopy, micropipettes, DNA extraction, PCR and DNA sequencing. Furthermore, I also gained knowledge with standard laboratory procedures such as data analysis such as calculating the amount of microorganisms and the species found throughout the Bishop Museum, this analysis was presented to researchers in the field and other Bishop Museum members with my PCMB intern group.

This research project resonates deeply with me because it integrates historical and modern microbiome analyses to explore how environmental factors have shaped the skin and gut microbiomes of microhylid frogs over 50 years. This project combines my love for museums, specifically the Bishop Museum, and research for micro biodiversity. As a future professor, investigating how different microhabitats (soil/leaf litter, shrubs, and streams) influence these communities will be significant topics of discussion for my future academic publications. Engaging in this project will not only enhance my technical and analytical skills, but also contribute significantly to my future research career and aspirations in academia.


## Training & Research Compliance (1500 character limit):

Alexis has joined the lab and participated in lab meetings. She completed the EHSO safety checklist on March 3, 2025 and copies of the paperwork are kept on file with the Butler Lab. During the training, hazards that may be present in the lab were described, among other topics covered by the checklist. 

Alexis has completed EHSO general laboratory safety training on March 1, 2025, as required by the IBC. All genetic and biological safety aspects of the project are covered by Dr. Butlerʻs IBC protocol, which will be renewed with Alexis added before the announcement of UROP awards. 

Project-specific training will be provided by Dr. Marguerite Butler beginning in March 2025. Dr. Butler is an expert in the methods described here and has protocol specific SOPs available for review prior to hands-on training. The lab has detailed emergency response and spill protocols that are reviewed before working in the lab. These discussions and training sessions have been ongoing since February 2025 to ensure all necessary compliance requirements are fully met by the start of the project.  


## Bibliography/References (5000 Character Limit)